### ISSUE TITLE
[1.3.0b2] Duplicate Pattern is misleading

### ISSUE THREAD

**Issue by brian3kb** [polyend/TrackerIssues#458]
### Steps to reproduce

When using the Duplicate Pattern option, the pattern is not actually duplicated into a new pattern, but the pattern content is doubled within the current pattern - so a 32 line pattern would be doubled to 64.

This is misleading, as I expected the pattern to be duplicated into a new pattern, so duplicating pattern 1, would create a copy of the pattern as pattern 2 (or the next available pattern number not already occupied).

As the maximum pattern length is currently 128 lines, patterns of more than 64 lines cannot be duplicated at all, instead, an error message appears stating the pattern is too long.

### Suggested Fix
If this is the desired functionality, then rename the button to 'Double Pattern', and disable the button if the pattern is too long instead of showing the error message.

Change the behavior to actually duplicate the pattern as a new pattern number (this is what I wanted to do originally, instead I used the Song mode to create a new pattern and copy over each track's patterns).

### Other important information

* How often does this bug happen?
 - Every time you use the duplicate pattern button.

* Does this bug happen in previous firmware versions?
 - Yes

* Does this bug happen in every project?
 - Yes

**Comment by Monsieur36** [comment]
I completely agree that this function should be _Double Pattern_ instead, but most importantly that we should have an actual direct "Duplicate Pattern" function i.e., without having to copy Pattern, switch to a new/blank Pattern, and paste (this method is clunky and interrupts looping playback due to the immediate/legato way Patterns switch).
